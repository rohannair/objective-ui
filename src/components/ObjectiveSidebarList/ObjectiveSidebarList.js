import React, { PropTypes } from 'react'
import styled from 'styled-components'
import FontIcon from 'react-toolbox/lib/font_icon'

const ObjectiveSidebarList = ({ list, className, getObjective }) => {
  const objectiveLock = ({ isPrivate }) => isPrivate ? <FontIcon value='lock' /> : null

  let children = list.map(o => (
    <div
      key={o.id}
      className="item"
      onClick={() => getObjective(o.id) }
    >
      { objectiveLock(o) }
      { o.name }
    </div>
  ))

  return (
    <div className={className}>
      { children }
    </div>
  )
}

export default styled(ObjectiveSidebarList)`
  .item {
    cursor: pointer;
    font-size: 0.85em;
    padding: 10px 20px;

    &:hover {
      background-color: rgba(255, 255, 255, 0.15);
    }
  }

  .material-icons {
    font-size: 1rem;
    padding-right: .2rem;
    position: relative;
    top: .2rem;
  }
`
