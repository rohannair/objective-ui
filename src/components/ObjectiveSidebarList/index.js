import ObjectiveSidebarList from './ObjectiveSidebarList'
export default ObjectiveSidebarList
