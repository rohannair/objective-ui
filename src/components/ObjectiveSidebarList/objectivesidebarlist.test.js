import React from 'react'

import ObjectiveSidebarList from './index.js'

describe('Test suite for ObjectiveSidebarList component', () => {
  it('ObjectiveSidebarList should exist', () => {
    let wrapper = shallow(<ObjectiveSidebarList/>)
    expect(wrapper).toMatchSnapshot()
  })
})

