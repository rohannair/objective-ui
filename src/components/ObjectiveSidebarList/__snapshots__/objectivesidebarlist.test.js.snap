exports[`Test suite for ObjectiveSidebarList component ObjectiveSidebarList should exist 1`] = `
<ObjectiveSidebarList
  className="HaMDi" />
`;
